### PR TITLE
Allows editor to change reportback status if they made a mistake

### LINKED ISSUE #4621: changing a file's status should change the rb's flagged status
once a reportback is flagged, it always remains flagged, even if an editor made a mistake in flagging it and then changed it to be approved, promoted, excluded, etc.

we should allow for editors to change their minds (provided obviously they didn't delete an image).

so when a file's status changes from flagged to promoted the reportback reportback_flagged should change from 1 to 0

### PR DESCRIPTION
**Pull request by itsjoekent**
Fixes #4621 

also forgot to configure git, author is screwed up. oops!

@angaither

### CODE REVIEW

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.admin.inc [hunk 1]
```diff
@@ -418,5 +418,3 @@ function dosomething_reportback_count_page() {
 
   return $output;
 }
-
-
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.admin.inc:422]: can you remove these changes?

**Comment by angaither** [comment]
Looks ok, just want to double check, when you flag something, then unflag it, the reportback `flagged` boolean goes from 1->0

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.admin.inc (hunk 1 above)
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.admin.inc:422]: looks like my editor auto did that, weird

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.admin.inc:422]: but I still have line 421 as a blank line. da fuq

**Comment by itsjoekent** [comment]
correct, that is what happens.

**Pull request merged**